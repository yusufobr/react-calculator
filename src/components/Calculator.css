td {
  text-align: center;
  padding-bottom: 2px;
  padding-right: 2px;
  width: 25%;
}

td:hover {
  background-color: #00000008;
}

input[type="text"] {
  height: 50px;
}

.operator {
  background-color: rgb(255, 146, 82);
}
