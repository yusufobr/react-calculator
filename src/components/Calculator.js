import React from 'react';
import './Calculator.css';

export default function Calculator() {
  return (
    <div className="text-[#333] text-sm font-bold p-12">
      <table className="h-[300px] w-[350px] mx-auto bg-[#E0E0E0]">
        <thead>
          <tr>
            <td colSpan="4">
              <input type="text" value="0" className="w-full pr-1 text-right text-xl font-normal text-white bg-[#858693]" />
            </td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <input type="button" value="AC" />
            </td>
            <td>
              <input type="button" value="+/-" />
            </td>
            <td>
              <input type="button" value="%" />
            </td>
            <td className="operator">
              <input type="button" value="รท" />
            </td>
          </tr>
          <tr>
            <td>
              <input type="button" value="7" />
            </td>
            <td>
              <input type="button" value="8" />
            </td>
            <td>
              <input type="button" value="9" />
            </td>
            <td className="operator">
              <input type="button" value="x" />
            </td>
          </tr>
          <tr>
            <td>
              <input type="button" value="4" />
            </td>
            <td>
              <input type="button" value="5" />
            </td>
            <td>
              <input type="button" value="6" />
            </td>
            <td className="operator">
              <input type="button" value="-" />
            </td>
          </tr>
          <tr>
            <td>
              <input type="button" value="1" />
            </td>
            <td>
              <input type="button" value="2" />
            </td>
            <td>
              <input type="button" value="3" />
            </td>
            <td className="operator">
              <input type="button" value="+" />
            </td>
          </tr>
          <tr>
            <td colSpan="2">
              <input type="button" value="0" />
            </td>
            <td>
              <input type="button" value="." />
            </td>
            <td className="operator">
              <input type="button" value="=" />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  );
}
