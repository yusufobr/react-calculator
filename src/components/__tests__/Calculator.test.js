import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('Calculator component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Calculator />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
