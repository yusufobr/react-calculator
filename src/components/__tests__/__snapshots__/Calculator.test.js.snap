// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component renders correctly 1`] = `
<div
  className="container mx-auto mt-16 gap-2 flex justify-between rounded-lg py-8 px-4 bg-[#ffffff80] drop-shadow-lg"
>
  <h2
    className="text-xl font-semibold capitalize"
  >
    let's do some math!
  </h2>
  <div
    className="text-[#333] text-sm font-bold "
  >
    <table
      className="h-[400px] w-[450px] mx-auto bg-[#E0E0E0] drop-shadow-lg"
    >
      <thead>
        <tr>
          <td
            colSpan="4"
          >
            <input
              className="w-full pr-1 text-right text-xl font-normal text-white bg-[#858693]"
              onChange={[Function]}
              type="text"
              value={0}
            />
          </td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="AC"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="+/-"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="%"
            />
          </td>
          <td
            className="operator"
          >
            <input
              onClick={[Function]}
              type="button"
              value="รท"
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="7"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="8"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="9"
            />
          </td>
          <td
            className="operator"
          >
            <input
              onClick={[Function]}
              type="button"
              value="x"
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="4"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="5"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="6"
            />
          </td>
          <td
            className="operator"
          >
            <input
              onClick={[Function]}
              type="button"
              value="-"
            />
          </td>
        </tr>
        <tr>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="1"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="2"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="3"
            />
          </td>
          <td
            className="operator"
          >
            <input
              onClick={[Function]}
              type="button"
              value="+"
            />
          </td>
        </tr>
        <tr>
          <td
            colSpan="2"
          >
            <input
              onClick={[Function]}
              type="button"
              value="0"
            />
          </td>
          <td>
            <input
              onClick={[Function]}
              type="button"
              value="."
            />
          </td>
          <td
            className="operator"
          >
            <input
              onClick={[Function]}
              type="button"
              value="="
            />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
